var/spack/repos/builtin/packages/elfutils/package.py

```diff
@@ -121,8 +122,10 @@ def configure_args(self):
         args.append("--with-zlib=%s" % spec["zlib"].prefix)
 
         if "+nls" in spec:
-            # configure doesn't use LIBS correctly
-            args.append("LDFLAGS=-Wl,--no-as-needed -L%s -lintl" % spec["gettext"].prefix.lib)
+            args.append(
+                "LDFLAGS=-Wl,--no-as-needed "
+                + " ".join(filter_system_paths(self.spec["gettext"].libs.ld_flags.split()))
```

Comment:
You may need the same care here that you applied in glib: avoid `-L` of system directories. It occurs to me that `gettext` could exclude system directories, e.g. `libs` can do:

```
def libs(self):
  libs = find_libraries(...)
  libs.directories = [] # you can set this after generating the object
  return libs
```

Reply:
Note: with respect to your suggestion to clear `libs.directories`, it is apparently habit to refer to `libs.directories[0]` without further checks. I could clear up the remaining uses of those, but it would create a fragility with respect to future recipe changes.